### ISSUE TITLE
Test suite failure

### ISSUE THREAD

**Issue by snoyberg** [MichaelXavier/cron#26]
```
      returns the first minute in the future that matches:               FAIL (8.09s)
        *** Failed! (after 19 tests and 1 shrink): 
        Exception:
          test/System/Test/Cron.hs:276:15-92: Irrefutable pattern failed for pattern Just actual
        CronSchedule * * * * *
        1858-11-17 13:50:59.579131947983 UTC
        Exception thrown while printing test case:
          test/System/Test/Cron.hs:276:15-92: Irrefutable pattern failed for pattern Just actual
        
        Use --quickcheck-replay '90 TFGenR 00000010C895C77300000000000F4240000000000000E0BC00000001FB8896C0 0 536869888 29 0' to reproduce.
      a schedule that produces Just for one t will produce it for any t: OK (0.11s)
        +++ OK, passed 100 tests.
```

**Comment by MichaelXavier** [comment]
@snoyberg Thanks for this. I should be able to carve out some time this weekend to look into this.

**Comment by MichaelXavier** [comment]
@snoyberg found the issue, it was only an issue in test code. Do you need this uploaded anywhere or is the test passing enough?

**Comment by snoyberg** [comment]
If you upload it to Hackage, then I can update Stackage to resume running
these test suites

**Comment by MichaelXavier** [comment]
I released the fix under version 0.4.1.1

**Comment by creichert** [comment]
Is there any plan to enable this in stackage: https://github.com/fpco/stackage/blob/e5e9d056a5a7caf2d05e148fc5f0aea26667b8f5/build-constraints.yaml#L1497

**Comment by MichaelXavier** [comment]
@creichert it looks like this is blocked on derive. In order for me to drop derive, I need [this pr](https://github.com/nick8325/quickcheck/pull/40) on quickcheck merged. We've been banging on the door there for months and it is unclear why nobody is responding.

**Comment by creichert** [comment]
@MichaelXavier Thanks, noted.

**Comment by MichaelXavier** [comment]
Note: I've released cron 0.4.2 that uses generics-sop instead of derive and is thus compatible with nightlies. I'll be PRing cron to be reenabled in stackage.
